File: android/app/src/main/java/noah/moohvie/models/Question.kt
package noah.moohvie.models

import java.util.UUID

data class AnswerOption(
    val id: UUID = UUID.randomUUID(),
    val label: String,
    val genresToAdd: List<Int> = emptyList(),
    val genresToExclude: List<Int> = emptyList(),
    val maxRuntime: Int? = null,
    val minVoteAverage: Double? = null,
    val minReleaseYear: Int? = null,
    val maxReleaseYear: Int? = null,
)

enum class QuestionPriority(val rank: Int) {
    ESSENTIAL(1), // dans le mode court (5 questions)
    IMPORTANT(2), // dans le mode moyen (12 questions)
    DETAILED(3),  // uniquement dans le mode long (20 questions)
}

data class Question(
    val id: UUID = UUID.randomUUID(),
    val text: String,
    val options: List<AnswerOption>,
    val priority: QuestionPriority,
)

File: Moohvie/Models/question.swift
import Foundation

struct AnswerOption: Identifiable {
    let id = UUID()
    let label: String
    let genresToAdd: [Int]
    let genresToExclude: [Int]

    var maxRuntime: Int? = nil
    var minVoteAverage: Double? = nil
    var minReleaseYear: Int? = nil
    var maxReleaseYear: Int? = nil

    init(
        label: String,
        genresToAdd: [Int] = [],
        genresToExclude: [Int] = [],
        maxRuntime: Int? = nil,
        minVoteAverage: Double? = nil,
        minReleaseYear: Int? = nil,
        maxReleaseYear: Int? = nil
    ) {
        self.label = label
        self.genresToAdd = genresToAdd
        self.genresToExclude = genresToExclude
        self.maxRuntime = maxRuntime
        self.minVoteAverage = minVoteAverage
        self.minReleaseYear = minReleaseYear
        self.maxReleaseYear = maxReleaseYear
    }
}

enum QuestionPriority: Int {
    case essential = 1   // dans le mode court (5 questions)
    case important = 2   // dans le mode moyen (12 questions)
    case detailed = 3    // uniquement dans le mode long (20 questions)
}

struct Question: Identifiable {
    let id = UUID()
    let text: String
    let options: [AnswerOption]
    let priority: QuestionPriority
}
